pkg/kube/kclient/client.go

```diff
@@ -90,6 +90,43 @@ func (n *informerClient[T]) Start(stopCh <-chan struct{}) {
 	n.startInformer(stopCh)
 }
 
+// internalIndex is the type we use to store a Kubernetes set of index, as well as the key to the index we care about.
+// Users should not interact with this directly, and should instead use CreateIndex.
+type internalIndex struct {
+	key     string
+	indexer cache.Indexer
+}
+
+func (i internalIndex) Lookup(key string) []interface{} {
+	res, err := i.indexer.ByIndex(i.key, key)
+	if err != nil {
+		// This should only happen if the key does not exist which should be impossible
```

Comment:
Could you clarify which key you are referring to? The key for the internal index or the passed `key`

Reply:
The internal index. Ill update the comment

Reply:
done